test/jest/cookieSyncManager.spec.ts

```diff
@@ -0,0 +1,454 @@
+                    getPersistence: () => ({testMPID: {
+                        csd: {}
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync(null, testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                'https://test.com%3Fredirect%3Dhttps%3A%2F%2Fredirect.com%26mpid%3DtestMPID',
+                5,
+                testMPID,
+                {},
+                undefined,
+                true,
+                false, 
+            );
+        });
+
+        // QUESTION: What is the purpose of this code path?
+        it('should call performCookieSync with mpid if previousMpid and mpid do not match', () => {
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [pixelSettings],
+                },
+                _Persistence: {
+                    getPersistence: () => ({testMPID: {
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync('other-mpid', testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                '',
+                5,
+                testMPID,
+                {},
+                undefined,
+                true,
+                false, 
+            );
+        });
+
+        it('should call performCookieSync with mpid and csd if previousMpid and mpid do not match', () => {
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [pixelSettings],
+                },
+                _Persistence: {
+                    getPersistence: () => ({testMPID: {
+                        csd: { 5: 1234567890 }
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync('other-mpid', testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                '',
+                5,
+                testMPID,
+                {
+                    5: 1234567890
+                },
+                undefined,
+                true,
+                false, 
+            );
+        });
+
+        // QUESTION: What is the purpose of this code path?
+        it('should call performCookieSync with mpid if previousMpid and mpid match', () => {
```

Comment:
As i mentioned in the long previous comment from the first question, seems like this code can be refactored to better represent when a cookie sync should happen.

1. when cookie sync has never happened yet (assuming it should based on whatever filters/privacy there are)
2. when cookie sync allotted time has lapsed

Reply:
The whole MPID equality portion of code can _probably_ be removed because we'd check for criteria 1 and 2 above for whatever the current MPID is

Reply:
Agreed. There's a lot of nested logic that could be broken out to make things more readable. I think it should likely be done as part of SQDSDKS-6915